test/README.md

```diff
@@ -0,0 +1,60 @@
+# Functions E2E Test
+
+## Lifecycle tests
+
+## Extended tests
+
+## On Cluster Builds tests
+
+On cluster builds e2e tests exercises functions built directly on cluster.
+The tests are organized per scenarios under `./_oncluster` folder.
+
+### Pre-requisites
+
+Prior to run On Cluster builds e2e tests ensure you are connected to
+a Kubernetes Cluster with the following deployed:
+
+- Knative Serving
+- Tekton
+- Tekton Tasks listed [here](../docs/reference/on_cluster_build.md)
+- Embedded Git Server (`func-git`) used by tests
+
+For your convenience you can run the following script to setup Tekton and required Tasks:
+```
+$ ../hack/tekton.sh
+```
+
+To install the Git Server required by tests, run:
+```
+$ ./gitserver.sh
+```
```

Comment:
Sure I can add it.. however my only concern I didn't added in the first time is, that it would be not required by all GH actions  jobs such as intergration tests, podman test, lifecycle tests etc, and that would save some time on CI checks to complete.

Reply:
Fair enough